interview-questions.md

```diff
@@ -0,0 +1,49 @@
+# ASSESSMENT 4: Interview Practice Questions
+
+Answer the following questions.
+
+First, without external resources. Challenge yourself to answer from memory.
+
+Then, research the question to expand on your answer. Even if you feel you have answered the question completely on your own, there is always something more to learn. Write your researched answer in your OWN WORDS.
+
+1. What is object-oriented programming? How is it different than functional programming?
+
+Your answer: Object oriented programming focuses its logic around objects as in Ruby, while functional focuses its logic in functions as in Javascript for example.
+
+Researched answer: Functional programming uses immutable data, while obejct oriented programming achieves results by altering the program's state.
+
+2. What is the difference between a Float and an Integer in Ruby?
+
+Your answer: Float is a decimal number while Integer is a whole number.
+
+Researched answer: Integers are whole numbers, Floats are numbers tha contain decimal point.
+
+3. Ruby has an implicit return. What does that mean?
+
+Your answer: It means it does not need the return word in order to run the logic.
+
+Researched answer: If a return is the last expression in a code execution, return is not needed.
```

Comment:
Implicit return means that even without actually putting down the word return, Ruby will return the last line of code on its own.